app.js

```diff
@@ -128,16 +128,88 @@ function createShipSquares(shipCoordinates, battleField)
     }
 }
 
-function createShips()
+function createShotAccess(battleField, computerShipCoordinates, userShipCoordinates)
 {
+    const shotPlaces = getArray(10, 10);
+    var userScore = 0;
+    var computerScore = 0;
+
+    battleField.addEventListener('click', doShot)
+
+    function doShot(e)
+    {
+        var pos = getPosition(e.target.classList[1]);
+        var shot = isHit(computerShipCoordinates[pos.x][pos.y]);
+        e.target.style.backgroundColor = shot.color;
+        userScore = hitCounter(userScore, shot.hit).score;
+        winner(userScore, '.player-side');
+        computerTurn();
+    }
+
+    function computerTurn()
+    {
```

Comment:
As I said to you on a call earlier:
this function deals with html elements and game logic - it is bad, also this function takes the state of the game from html elements (sample - isHit(color))

How to improve the code?
Assume you develop a console version of BattleShip game. let's say you have the object 'game' which has functions to perform hit, check the state of any piece of battlefield, check the state of the game (who wins)
then your battleField.addEventListener('click', ) - let's say should call functions or 'game' object and update the state of HTML elements.

So, try to create a console version of the BattleShip game, and later connect it to the HTML elements. This should resolve most of the code issues.